examples/improved_wgan.py

```diff
@@ -72,7 +73,7 @@ def gradient_penalty_loss(y_true, y_pred, trainable_weights):
     y_true and y_pred to loss functions. To get around this, we make a partial() of the function with the
     trainable_weights argument, and use that for model training."""
     averaged_samples_loss = K.mean(y_pred)
-    gradients = K.gradients(averaged_samples_loss, trainable_weights)
+    gradients = K.gradients(averaged_samples_loss, averaged_samples)

```

Comment:
I think this should be `gradients = K.gradients(y_pred, averaged_samples)` "since we penalize the norm of
the critic’s gradient with respect to each input independently, and not the entire batch". In other words you're calculating the gradient of the output of the discriminator (not a loss computed on the output which averages across a batch).